pandas/tests/indexes/multi/test_duplicates.py

```diff
@@ -299,6 +300,54 @@ def test_duplicated_drop_duplicates():
     tm.assert_index_equal(idx.drop_duplicates(keep=False), expected)
 
 
+@pytest.mark.parametrize(
+    "array,expected,dtype",
+    [
+        (
+            [
+                np.nan + np.nan * 1j,
+                0,
+                1j,
+                1j,
+                1,
+                1 + 1j,
+                1 + 2j,
+                1 + 1j,
+                np.nan,
+                np.nan + np.nan * 1j,
+            ],
+            Series(
+                [False, False, False, True, False, False, False, True, False, True],
+                dtype=bool,
+            ),
+            np.complex64,
+        ),
+        (
+            [
+                np.nan + np.nan * 1j,
+                0,
+                1j,
+                1j,
+                1,
+                1 + 1j,
+                1 + 2j,
+                1 + 1j,
+                np.nan,
+                np.nan + np.nan * 1j,
+            ],
+            Series(
+                [False, False, False, True, False, False, False, True, False, True],
+                dtype=bool,
+            ),
+            np.complex128,
+        ),
+    ],
+)
```

Comment:
looks like `array` and `expected` are the same in both cases; better to define them inside the test and just parameterize over the dtype